client/signup/steps/site-topic/index.jsx

```diff
@@ -26,23 +28,36 @@ class SiteTopicStep extends Component {
 		setSiteTopic: PropTypes.func.isRequired,
 		signupProgress: PropTypes.array,
 		stepName: PropTypes.string,
+		siteTopic: PropTypes.string,
 	};
 
 	state = {
-		siteTopic: '',
+		siteTopicInputValue: '',
 	};
 
-	onChangeTopic = event => this.setState( { siteTopic: event.target.value } );
+	componentDidMount() {
```

Comment:
This is interesting - are there any pros/cons in initializing state in `componentDidMount `? (asking because I don't know :)).
Say, versus initializing in the constructor?

```
constructor( props ) {
		super( props );
		this.state = {
			siteTopicInputValue: props.siteTopic,
		};
	}
```

Reply:
Well, the timing in the whole component lifecycle is not the same. If you want to make sure every time the state value is initialized as the prop value on mounting, it would be better to initialize it in a `componentDidMount()` handler since it is possible to remount a component without destroying it from the heap. However, I don't see there is any difference in this case and putting it in the constructor might be clearer, so I've moved the logic there.